subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -249,6 +249,20 @@ In precompiled script plugins, type safe Kotlin DSL accessor generation now fail
 
 Starting in Gradle 7.6, builds could enable this behavior with the `org.gradle.kotlin.dsl.precompiled.accessors.strict` system property. This behavior is now default. The property has been deprecated and its usage should be removed. You can find more information about this property <<upgrading_version_7.adoc#strict-kotlin-dsl-precompiled-scripts-accessors, below>>.
 
+==== Init scripts are applied to `buildSrc` builds
+
+Init script specified using `--init-script` are now applied to `buildSrc` builds. In previous releases these were applied to included builds but not `buildSrc builds.
+This behaviour is now consistent for `buildSrc` and included builds.
+
+==== Gradle no longer runs the `build` task for `buildSrc` builds
```

Comment:
We could keep this behavior in 8.x. Since we inject build configuration into buildSrc, we could make the jar variant depend on build.

Reply:
Do we want to keep it though? I think it's a good thing that `build` doesn't run.

Reply:
I was thinking that this change had the biggest obvious impact and so we might want to be more conservative. I remember someone abusing buildSrc to attach random other tasks as a dependency to `build` to initialize other things. 

I think as long as we suggest a reasonable way to get back to the old behavior (e.g., run with `:buildSrc:build` on CI), we should be OK. We shouldn't suggest something like `jar.dependsOn test`